python/feather/tests/test_reader.py

```diff
@@ -287,6 +287,18 @@ def test_read_columns(self):
         df = pd.DataFrame(data)
         expected = pd.DataFrame({c:data[c] for c in columns})
         self._check_pandas_roundtrip(df, expected, columns=columns)
+
+    def test_overwritten_file(self):
+        path = random_path()
+
+        num_values = 100
+        np.random.seed(0)
+
+        values = np.random.randint(0, 10, size=num_values)
+        feather.write_dataframe(pd.DataFrame({'ints':values}), path)
+
+        df = pd.DataFrame({'ints':values[0:num_values//2]})
+        self._check_pandas_roundtrip(df, path=path)
```

Comment:
Yes it does fail without the fix cf. just-created build with tests only https://travis-ci.org/guillett/feather/builds/170437697.